paddle/fluid/framework/ir/memory_optimize_pass/reference_count_pass.cc

```diff
@@ -344,47 +305,45 @@ void ReferenceCountPass::ApplyImpl(ir::Graph *graph) const {
 
       PADDLE_ENFORCE_EQ(var_desc->Name(), var_name);
 
-      for (auto iter = var_handles.rbegin(); iter != var_handles.rend();
-           ++iter) {
-        if ((*iter)->Node()->IsCtrlVar()) {
-          break;
-        }
+      PADDLE_ENFORCE_EQ(
+          var_handles.empty(), false,
+          platform::errors::InvalidArgument("Variable %s not found", var_name));
+      auto last_ver_var = var_handles.back();
 
-        VLOG(10) << "Try to find last living ops of " << var_name << " "
-                 << (iter - var_handles.rbegin()) << " time";
-        LastLiveOpSearchStatus status = LastLiveOpSearchStatus::kFailure;
-        auto result = ExtractComputationOpFromLastLivedVar(
-            *iter, i, var_name, shrink_func, &status);
-
-        // Seldomly, some vars may have no pending or preceding computation ops
-        // Just break;
-        if (status == LastLiveOpSearchStatus::kFailure) {
-          VLOG(1) << "Cannot find last live ops of variable " << var_name
-                  << " in scope " << (*iter)->scope_idx();
-          break;
-        }
+      if (last_ver_var->Node()->IsCtrlVar()) {
+        continue;
+      }
 
-        if (status == LastLiveOpSearchStatus::kShouldPrecede) {
-          VLOG(10) << "Try to precede reference count computing at var "
-                   << var_name;
-          continue;
-        }
+      LastLiveOpSearchStatus status = LastLiveOpSearchStatus::kFailure;
+      auto result = ExtractComputationOpFromLastLivedVar(
+          last_ver_var, i, var_name, shrink_func, &status);
+
+      // Seldomly, some vars may have no pending or preceding computation ops
+      // Just break;
+      if (status == LastLiveOpSearchStatus::kFailure) {
+        VLOG(1) << "Cannot find last live ops of variable " << var_name
+                << " in scope " << last_ver_var->scope_idx();
+        continue;
+      }
 
-        PADDLE_ENFORCE_EQ(status, LastLiveOpSearchStatus::kSuccess);
-        PADDLE_ENFORCE(!result.empty(), "Last living ops of %s cannot be empty",
-                       var_name);
+      PADDLE_ENFORCE_EQ(status, LastLiveOpSearchStatus::kSuccess,
+                        platform::errors::InvalidArgument(
+                            "status must be success or faiulre"));
```

Comment:
Maybe "status must be success here." is more appropriate, since `kFailure ` is already handled before.

Reply:
Done.